Spoj1Divisor Summation/Program.cs

```diff
@@ -7,26 +7,85 @@
 
 namespace Spoj1Divisor_Summation
 {
-    public class Program
-    {
-        public static void Main()
-        {
-            var reader = new StreamReader(Console.OpenStandardInput());
-            var writer = new StreamWriter(Console.OpenStandardOutput());
-            writer.AutoFlush = false;
-
-            DivisorSumation(reader, writer);
+    public class Program
+    {
+        public static void Main()
+        {
+            try
+            {
+                Console.WriteLine("Input:");
+                var reader = new StringReader(NumberInput());
+
+                Console.WriteLine(" \nOutput:");
+                var writer = new StreamWriter(Console.OpenStandardOutput());
+                writer.AutoFlush = false;
+
+                DivisorSumation(reader, writer);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+            }        
+            Console.ReadKey();
         }
 
         public static void DivisorSumation(TextReader reader, TextWriter writer)
-        {
-           //*******************************
-           //* TODO: Implement method body *
-           //*******************************
+        {           
+            int numberCount = Convert.ToInt32(reader.ReadLine());
+
+            for (int i = 0; i < numberCount; i++)
+            {            
+                int number = Convert.ToInt32(reader.ReadLine());
+                List<int> divisors = CalculateDivisors(number);         
+                writer.WriteLine(divisors.Sum());
+            }
 
             reader.Close();
             writer.Flush();
             writer.Close();
         }
+
+        private static List<int>CalculateDivisors(int number)
+        {
+            List<int> divisors = new List<int>();
+
+            for (int i = 1; i < number; i++)
+                if ((number % i) == 0) divisors.Add(i);
+
+            return divisors;
+        }
+
+
+        public static string NumberInput()

```

Comment:
Dodatkowo komentarz w miejsu wykorzystania metody. 

"Chciałbym żeby reader przyjmował "Console.OpenStandardInput()", a jeśli potrzebujesz operować na konsoli, to operuj na zmiennej 'reader'"